alb/terraform-examples/cluster-deployment/cluster-deployment.tf

```diff
@@ -0,0 +1,175 @@
+terraform {
+  required_providers {
+    vsphere = {
+      source  = "hashicorp/vsphere"
+      version = "2.0.2"
+     }    
+    avi = {
+      source  = "vmware/avi"
+      version = "21.1.2"
+    }
+  }
+}
+
+provider "vsphere" {
+  user                   = var.vsphere_user
+  password               = var.vsphere_password
+  vsphere_server         = var.vsphere_server
+  allow_unverified_ssl   = true
+}
+
+
+data "vsphere_resource_pool" "resource_pool" {
+  name                  = var.vsphere_resource_pool
+  datacenter_id 	= "${data.vsphere_datacenter.datacenter.id}"
+}
+
+data "vsphere_content_library" "content_library" {
+  name            = "AVI"
+}
+
+data "vsphere_content_library_item" "controllerovf" {
+  name        = "Controller"
+  type	      = "ovf"
+  library_id  = data.vsphere_content_library.content_library.id
+}
+
+resource "vsphere_virtual_machine" "controller" {
+  name              = var.Controller_vm_name[count.index]
+  resource_pool_id  = data.vsphere_resource_pool.resource_pool.id
+  datastore_id      =  "${data.vsphere_datastore.datastore.id}"
+  count             = 3
+  num_cpus          = 8
+  memory            = 24576
+  folder            = var.vm_folder
+  scsi_type         = "lsilogic"
+
+  network_interface {
+    network_id   = "${data.vsphere_network.management_net.id}"
+  }
+  lifecycle {
+    ignore_changes = [guest_id]
+  }
+  disk {
+    label            = "disk0"
+    size             = 128
+    thin_provisioned = false
+  }
+  clone {
+   template_uuid = data.vsphere_content_library_item.controllerovf.id
+  }
+  vapp {
+    properties = {
+      "mgmt-ip"     = var.avi_ctrl_mgmt_ips[count.index]
+      "mgmt-mask"   = var.subnetMask
+      "default-gw"  = var.defaultGateway
+    }
+  }
+  wait_for_guest_net_timeout = 10   # Increase in slow environments
+}
+
+resource "null_resource" "wait_https_controllers" {
+  depends_on = [vsphere_virtual_machine.controller]
+  count = 3
+
+  provisioner "local-exec" {
+    command = "count=1 ; while [ $(curl -k -s -o /dev/null -w \"%%{http_code}\"   https://${var.avi_ctrl_mgmt_ips[count.index]}) != \"200\" ]; do echo \"Attempt $count: Waiting for Avi Controllers to be ready...\"; sleep 5 ; count=$((count+1)) ;  if [ \"$count\" = 120 ]; then echo \"ERROR: Unable to connect to Avi Controller API\" ; exit 1 ; fi ; done"
+  }
+}
+
+
+provider "avi" {
+  avi_username   = var.avi_username
+  avi_password   = "58NFaGDJm(PJH0G"
```

Comment:
`var.avi_password` ?

Reply:
This is the original password at time of deployment. I put it in the script as I felt it wasnt needed in the input variables file.

Reply:
I would place it in a `terraform.tfvars.example` file. See the PCA examples.

Reply:
Its the default password it is needed for this to connect. I dont see the problem with leaving it here. It is publicly known default password. We have it as a note when downloading the ova.

Reply:
I added it to the variables file.